### PR TITLE
Admin

### PR DESCRIPTION
**Pull request by Intrast**
**STATUS**
READY

**DESCRIPTION**

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [x] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
default

default

default

default

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -11,4 +11,42 @@ def stop_impersonating
...
+  end
+
+ private
+
+ def users_params
+   params.require(:user).permit(:first_name, :last_name, :role, :department, :about, :avatar, :job_role)
+ end
+
+ def resource
+   User.find_by(id: params[:id])
```
Comment (igorkasyanchuk) [inline app/controllers/admin/users_controller.rb]: чому find_by

File: app/views/admin/users/_form_edit.html.slim [hunk 2]
```diff
@@ -0,0 +1,11 @@
+= simple_form_for [:admin, user] do |f|
+  = f.input :avatar, label: "Select your profile avatar", wrapper: :upload_file
+  = f.input :first_name, wrapper: :bulma_input, required: true, label_html: { class: "has-text-left"}
+  = f.input :last_name, wrapper: :bulma_input, required: true, label_html: { class: "has-text-left"}
+  = f.input :job_role, wrapper: :bulma_input, label_html: { class: "has-text-left"}
+  = f.input :department, wrapper: :bulma_input, label_html: { class: "has-text-left"}
+  = f.input :about, wrapper: :textarea, label_html: { class: "has-text-left"}
+  = f.label :role, class: 'label has-text-left'
```
Comment (igorkasyanchuk) [inline app/views/admin/users/_form_edit.html.slim]: в сімпл формі нема лейблів
має генеруватись враппером
правте враппер

File: app/views/admin/users/edit.html.slim [hunk 3]
```diff
@@ -0,0 +1,7 @@
+- title "Profile Settings"
+.container.has-text-centered
+  .column.is-4.is-offset-4.box.profile
+    .title Profile Settings
+    figure.container.image.is-64x64
+      = profile_avatar(@user)
+    = render '/admin/users/form_edit', user: @user
```
Comment (igorkasyanchuk) [inline app/views/admin/users/edit.html.slim]: просто form має бути

File: app/views/admin/users/index.html.slim [hunk 4]
```diff
@@ -0,0 +1,20 @@
+- title 'All users'
+.container
+  .level
+    .level-left
+      .title
+        | All users
+  .index-margin
+    - @users.each do |user|
+      - if !user.admin?
```
Comment (igorkasyanchuk) [inline app/views/admin/users/index.html.slim]: поганий код що це за перевірка
хочеш виводити адмінів - ДІСТАВАЙ ТІЛЬКИ АДМІНІВ

File: app/views/admin/users/index.html.slim [hunk 5]
```diff
@@ -0,0 +1,20 @@
+- title 'All users'
+.container
+  .level
+    .level-left
+      .title
+        | All users
+  .index-margin
+    - @users.each do |user|
```
Comment (AndriySolonyna) [inline app/views/admin/users/index.html.slim:8]: add scope none_admins

File: spec/controllers/admin/users_controller_spec.rb [hunk 6]
```diff
@@ -40,4 +42,58 @@
       end
     end
   end
+
+  describe "if user admin" do
```
Comment (AndriySolonyna) [inline spec/controllers/admin/users_controller_spec.rb]: http://www.betterspecs.org/#contexts

File: app/views/admin/users/_form.html.slim [hunk 7]
```diff
@@ -0,0 +1,11 @@
+= simple_form_for [:admin, user] do |f|
+  = f.input :avatar, label: "Select your profile avatar", wrapper: :upload_file
+  = f.input :first_name, wrapper: :bulma_input, required: true, label_html: { class: "has-text-left"}
+  = f.input :last_name, wrapper: :bulma_input, required: true, label_html: { class: "has-text-left"}
+  = f.input :job_role, wrapper: :bulma_input, label_html: { class: "has-text-left"}
+  = f.input :department, wrapper: :bulma_input, label_html: { class: "has-text-left"}
+  = f.input :about, wrapper: :textarea, label_html: { class: "has-text-left"}
+  = f.input :role, collection: ["admin", "user"], include_blank: false,
```
Comment (igorkasyanchuk) [inline app/views/admin/users/_form.html.slim]: use constants from model, why hard coded values here

**Review by igorkasyanchuk — APPROVED** [review]

File: app/assets/stylesheets/user.scss [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    margin-bottom: 0;
+  }
+}
+
+.button-margin {
+  margin-top: 37px;
+}
+
+.column {
+  .select:not(.is-multiple):not(.is-loading)::after {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/user.scss]: THIS IS NOT GOOD
give CLASS to select and specify this style

because you with this change affected ALL selects

**Pull request merged**